### ISSUE TITLE
"Index 0 out of bounds for length 0" Error when starting a postgres-11 container

### ISSUE THREAD

**Issue by ghevge** [fabric8io/docker-maven-plugin#1511]
While setting up the containers for my integrations tests, I'm starting 2 containers: a redis and a postgres-11. The Redis container is starting ok, while the postres-11 one is failing with the weird error below.

Any idea what could cause this problem?

Thanks

I'm using openjdk 11

```
# lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description: Ubuntu 20.04.3 LTS
Release: 20.04
Codename: focal

```

maven definition:

```
<plugin>
          <groupId>io.fabric8</groupId>
          <artifactId>docker-maven-plugin</artifactId>
          <version>${docker.maven.plugin.version}</version>
          <executions>
              <execution>
                  <id>prepare-it-database</id>
                  <phase>pre-integration-test</phase>
                  <goals>
                      <goal>start</goal>
                  </goals>
                  <configuration>
                      <images>
                            <image>
                              <name>my-repo.com/docker.io/redis:3.2</name>
                              <alias>it-redis</alias>
                              <run>
                                  <ports>
                                      <port>6377:6379</port>
                                  </ports>
                                  <hostname>it-redis</hostname>
                                  <wait>
                                      <log>(?s)The server is now ready to accept connections</log>
                                      <time>20000</time>
                                  </wait>
                                  <log>
                                      <prefix>REDIS-TEST</prefix>
                                      <date>ISO8601</date>
                                      <color>yellow</color>
                                  </log>
                              </run>
                          </image>
                          <image>
                              <name>my-repo.com/docker.io/postgres:11-alpine</name>
                              <alias>it-database</alias>
                              <run>
                                  <env>
                                      <POSTGRES_PASSWORD>postgres</POSTGRES_PASSWORD>
                                  </env>
                                  <ports>
                                      <port>it-database.port:5432</port>
                                  </ports>
                                  <hostname>test-db</hostname>
                                  <wait>
                                      <log>(?s)database system is ready to accept connections.*database system is ready to accept connections</log>
                                      <time>20000</time>
                                  </wait>
                                  <log>
                                      <prefix>POSTGRESQL-TEST</prefix>
                                      <date>ISO8601</date>
                                      <color>blue</color>
                                  </log>
                              </run>
                          </image>
                      </images>
                  </configuration>
              </execution>
              <execution>
                  <id>remove-it-database</id>
                  <phase>post-integration-test</phase>
                  <goals>
                      <goal>stop</goal>
                  </goals>
              </execution>
          </executions>
      </plugin>
```

```
[INFO] DOCKER> [my-repo.com/docker.io/redis:3.2] "it-redis": Waited on log out '(?s)The server is now ready to accept connections' 2069 ms
[DEBUG] DOCKER> Container create config: {"Image":"my-repo.com/docker.io/postgres:11-alpine","Hostname":"test-db","ExposedPorts":{"5432/tcp":{}},"Env":["POSTGRES_PASSWORD=postgres"],"Labels":{"dmp.coordinates":"com.bla.test:test-api-component:0.0.0-localdev-SNAPSHOT"},"HostConfig":{"PortBindings":{"5432/tcp":[{"HostPort":""}]}}}
[INFO] DOCKER> [my-repo.com/docker.io/postgres:11-alpine] "it-database": Start container 495fb2946271
[ERROR] DOCKER> Error occurred during container startup, shutting down...
[DEBUG] DOCKER> shutdown will wait max of 0 seconds before removing container
[INFO] DOCKER> [my-repo.com/docker.io/postgres:11-alpine] "it-database": Stop and removed container 495fb2946271 after 0 ms
[DEBUG] DOCKER> shutdown will wait max of 0 seconds before removing container
2021-12-17T13:25:10.069-05:00 REDIS-TEST1:signal-handler (1639765510) Received SIGTERM scheduling shutdown...
2021-12-17T13:25:10.081-05:00 REDIS-TEST1:M 17 Dec 18:25:10.081 # User requested shutdown...
2021-12-17T13:25:10.081-05:00 REDIS-TEST1:M 17 Dec 18:25:10.081 * Saving the final RDB snapshot before exiting.
2021-12-17T13:25:10.086-05:00 REDIS-TEST1:M 17 Dec 18:25:10.086 * DB saved on disk
2021-12-17T13:25:10.086-05:00 REDIS-TEST1:M 17 Dec 18:25:10.086 # Redis is now ready to exit, bye bye...
[INFO] DOCKER> [my-repo.com/docker.io/redis:3.2] "it-redis": Stop and removed container 8163d4fcaebf after 0 ms
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  11:40 min
[INFO] Finished at: 2021-12-17T13:25:10-05:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.fabric8:docker-maven-plugin:0.33.0:start (prepare-it-database) on project test-api-component: Execution prepare-it-database of goal io.fabric8:docker-maven-plugin:0.33.0:start failed: Index 0 out of bounds for length 0 -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal io.fabric8:docker-maven-plugin:0.33.0:start (prepare-it-database) on project test-api-component: Execution prepare-it-database of goal io.fabric8:docker-maven-plugin:0.33.0:start failed: Index 0 out of bounds for length 0
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.apache.maven.wrapper.BootstrapMainStarter.start (BootstrapMainStarter.java:39)
    at org.apache.maven.wrapper.WrapperExecutor.execute (WrapperExecutor.java:122)
    at org.apache.maven.wrapper.MavenWrapperMain.main (MavenWrapperMain.java:61)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution prepare-it-database of goal io.fabric8:docker-maven-plugin:0.33.0:start failed: Index 0 out of bounds for length 0
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:148)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.apache.maven.wrapper.BootstrapMainStarter.start (BootstrapMainStarter.java:39)
    at org.apache.maven.wrapper.WrapperExecutor.execute (WrapperExecutor.java:122)
    at org.apache.maven.wrapper.MavenWrapperMain.main (MavenWrapperMain.java:61)
Caused by: java.lang.IndexOutOfBoundsException: Index 0 out of bounds for length 0
    at jdk.internal.util.Preconditions.outOfBounds (Preconditions.java:64)
    at jdk.internal.util.Preconditions.outOfBoundsCheckIndex (Preconditions.java:70)
    at jdk.internal.util.Preconditions.checkIndex (Preconditions.java:248)
    at java.util.Objects.checkIndex (Objects.java:372)
    at java.util.ArrayList.get (ArrayList.java:459)
    at com.google.gson.JsonArray.get (JsonArray.java:194)
    at io.fabric8.maven.docker.model.ContainerDetails.createPortBindings (ContainerDetails.java:200)
    at io.fabric8.maven.docker.model.ContainerDetails.getPortBindings (ContainerDetails.java:139)
    at io.fabric8.maven.docker.service.RunService.updateMappedPortsAndAddresses (RunService.java:461)
    at io.fabric8.maven.docker.service.RunService.createAndStartContainer (RunService.java:156)
    at io.fabric8.maven.docker.service.helper.StartContainerExecutor.startContainers (StartContainerExecutor.java:47)
    at io.fabric8.maven.docker.StartMojo.lambda$startImage$0 (StartMojo.java:300)
    at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly (TrustedListenableFutureTask.java:117)
    at com.google.common.util.concurrent.InterruptibleTask.run (InterruptibleTask.java:38)
    at com.google.common.util.concurrent.TrustedListenableFutureTask.run (TrustedListenableFutureTask.java:77)
    at java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:515)
    at java.util.concurrent.FutureTask.run (FutureTask.java:264)
    at com.google.common.util.concurrent.MoreExecutors$DirectExecutorService.execute (MoreExecutors.java:260)
    at java.util.concurrent.ExecutorCompletionService.submit (ExecutorCompletionService.java:184)
    at io.fabric8.maven.docker.StartMojo.startImage (StartMojo.java:298)
    at io.fabric8.maven.docker.StartMojo.executeInternal (StartMojo.java:166)
    at io.fabric8.maven.docker.AbstractDockerMojo.execute (AbstractDockerMojo.java:238)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.apache.maven.wrapper.BootstrapMainStarter.start (BootstrapMainStarter.java:39)
    at org.apache.maven.wrapper.WrapperExecutor.execute (WrapperExecutor.java:122)
    at org.apache.maven.wrapper.MavenWrapperMain.main (MavenWrapperMain.java:61)
[ERROR] 
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException

```

**Comment by qiuyanfeng** [comment]
Updated docker.maven.plugin.version to 0.38.1 fixed this issue.

**Comment by rhuss** [comment]
@ghevge can you confirm that the issue is fixed for you, too, with 0.38.1 ? If so, please feel free to close this issue. thanks!
